daphne_worker_test/src/utils.rs

```diff
@@ -13,3 +16,67 @@ cfg_if! {
         pub fn set_panic_hook() {}
     }
 }
+
+/// LogWriter helps us write to the worker console.
+///
+/// It provides and io::Write implementation that provides line
+/// buffering.  It also takes care of emitting a timestamp as the
+/// timestamp code in the tracing library tries to use standard
+/// clock code, which does not work in a worker.
+struct LogWriter {
+    buffer: String,
+}
+
+impl LogWriter {
+    fn new() -> Self {
+        LogWriter {
+            buffer: String::new(),
+        }
+    }
+}
+
+impl io::Write for LogWriter {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        // TODO(bhalley)  This is not very efficient, but it is simple!
```

Comment:
Why is this inefficient?

Reply:
"inefficient" might have been too strong :).   It always accumulates the buffer into the pending output buffer, and then checks if the chunk being written ends a line.  A more optimized bit of code might deal with the common case of a whole line being written in one chunk by avoiding the extra copying.  Another simplification in this code is it assumes that whatever is generating things is also line buffering, e.g. that we're not going to get a chunk like "something\nsomething else" (without the final \n")